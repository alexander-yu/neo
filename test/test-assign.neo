void main() {
    var string s;
    create matrix<int> a = [[1, 2], [3, 4]];
    create array<int> b = {| 1, 2, 3 |};

    s = "Hello!\n";
    print(s);

    print(a[0, 0] = 5);
    print("\n");
    print(a);
    print("\n");

    a[:, :] = [[3, 4], [5, 6]];
    print(a);
    print("\n");
    print(b[0] = 4);
    print("\n");
    print(b);
    print("\n");

    b[1:] = {| 4, 5 |};
    print(b);
}