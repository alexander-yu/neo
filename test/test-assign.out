Hello!
5
[[5, 2],
 [3, 4]]
[[3, 4],
 [5, 6]]
4
{|4, 2, 3|}
{|4, 4, 5|}
